miso-web/src/main/webapp/scripts/download/download.js

```diff
@@ -0,0 +1,172 @@
+            mimeType = strMimeType || defaultMime,
+            payload = data,
+            url = !strFileName && !strMimeType && payload,
+            anchor = document.createElement("a"),
+            toString = function(a){return String(a);},
+            myBlob = (self.Blob || self.MozBlob || self.WebKitBlob || toString),
+            fileName = strFileName || "download",
+            blob,
+            reader;
+            myBlob= myBlob.call ? myBlob.bind(self) : Blob ;
+      
+        if(String(this)==="true"){ //reverse arguments, allowing download.bind(true, "text/xml", "export.xml") to act as a callback
+            payload=[payload, mimeType];
+            mimeType=payload[0];
+            payload=payload[1];
+        }
+
+
+        if(url && url.length< 2048){ // if no filename and no mime, assume a url was passed as the only argument
+            fileName = url.split("/").pop().split("?")[0];
+            anchor.href = url; // assign href prop to temp anchor
+            if(anchor.href.indexOf(url) !== -1){ // if the browser determines that it's a potentially valid url path:
+                var ajax=new XMLHttpRequest();
+                ajax.open( "GET", url, true);
+                ajax.responseType = 'blob';
+                ajax.onload= function(e){ 
+                  download(e.target.response, fileName, defaultMime);
+                };
+                setTimeout(function(){ ajax.send();}, 0); // allows setting custom ajax headers using the return:
+                return ajax;
+            } // end if valid url?
+        } // end if url?
+
+
+        //go ahead and download dataURLs right away
+        if(/^data:([\w+-]+\/[\w+.-]+)?[,;]/.test(payload)){
+        
+            if(payload.length > (1024*1024*1.999) && myBlob !== toString ){
+                payload=dataUrlToBlob(payload);
+                mimeType=payload.type || defaultMime;
+            }else{          
+                return navigator.msSaveBlob ?  // IE10 can't do a[download], only Blobs:
+                    navigator.msSaveBlob(dataUrlToBlob(payload), fileName) :
+                    saver(payload) ; // everyone else can save dataURLs un-processed
+            }
+            
+        }else{//not data url, is it a string with special needs?
+            if(/([\x80-\xff])/.test(payload)){            
+                var i=0, tempUiArr= new Uint8Array(payload.length), mx=tempUiArr.length;
+                for(i;i<mx;++i) tempUiArr[i]= payload.charCodeAt(i);
+                payload=new myBlob([tempUiArr], {type: mimeType});
+            }         
+        }
+        blob = payload instanceof myBlob ?
+            payload :
+            new myBlob([payload], {type: mimeType}) ;
+
+
+        function dataUrlToBlob(strUrl) {
+            var parts= strUrl.split(/[:;,]/),
+            type= parts[1],
+            indexDecoder = strUrl.indexOf("charset")>0 ? 3: 2,
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This check ignores index 0; consider using 'includes' method to make this check safe and explicit. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS2692)